### PR TITLE
Added test_ssh_after_deleting_floating for neutron.

### PR DESCRIPTION
**Pull request by alexgrom9393**
Added test_ssh_after_deleting_floating for neutron.
Added SSHClient with conditional support of sftp.

### CODE REVIEW

File: mos_tests/environment/ssh.py [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+        return self._ssh.connect(
+            self.host, port=self.port, username=self.username,
+            password=self.password)
+
+    def reconnect(self):
+        self._ssh = paramiko.SSHClient()
+        self._ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+        self.connect()
+        if self.use_sftp:
+            self._sftp = self._ssh.open_sftp()
```
Comment (gdyuldin) [inline mos_tests/environment/ssh.py]: May be better make _sftp property like this:

```
@property
def _sftp(self):
    if self._sftp_client is None:
        self._sftp_client = self._ssh.open_sftp()
    return self._sftp_client
```

In such case it won't initialized until it needed.

File: mos_tests/neutron/python_tests/base.py [hunk 2]
```diff
@@ -141,3 +178,29 @@ def check_vm_is_connectable(self, vm):
...
+
+        def is_accessible():
+            try:
+                with self.env.get_ssh_to_cirros(vm_ip, pkeys) as vm_remote:
+                    vm_remote.execute("date")
+                    return True
+            except Exception:
+                return False
+
+        wait(lambda: is_accessible(),
```
Comment (gdyuldin) [inline mos_tests/neutron/python_tests/base.py]: I've replaced devops wait with waiting wait (https://pypi.python.org/pypi/waiting). So this won't work after merge. You should rename args of wait call

File: mos_tests/neutron/python_tests/test_floating_ip.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    def check_vm_inaccessible_by_ssh(self, vm_ip, pkeys):
+        """Check that instance is inaccessible with ssh via floating_ip.
+
+        :param vm_ip: floating_ip of instance
+        :param pkeys: ip of instance to ping
+        """
+        with pytest.raises(NoValidConnectionsError) as exc:
+            with self.env.get_ssh_to_cirros(vm_ip, pkeys) as vm_remote:
+                vm_remote.execute("date")
+        assert "Unable to connect to port 22 on  or {}".format(vm_ip) \
```
Comment (gdyuldin) [inline mos_tests/neutron/python_tests/test_floating_ip.py]: Maybe smth should be between "on" and "or"

File: mos_tests/neutron/python_tests/test_floating_ip.py [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+            'Floatingip is not in the DOWN state.' \
+            'disassociate_floating_ip has failed.'
+
+        assert res2 is None, 'SSH hasn\'t been stopped'
+
+        # check that vm became inaccessible with ssh
+        self.check_vm_inaccessible_by_ssh(vm_ip=ip, pkeys=pkeys)
+
+        # for cleanup delete floating_ip using neutron
+        self.os_conn.delete_floating_ip(self.floating_ip, use_neutron=True)
```
Comment (gdyuldin) [inline mos_tests/neutron/python_tests/test_floating_ip.py]: This should be in fixture

File: mos_tests/neutron/python_tests/test_floating_ip.py [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+        res2 = None
+
+        with pytest.raises(SSHException):
+            with self.env.get_ssh_to_cirros(ip, pkeys) as vm_remote:
+                res1 = vm_remote.execute("ping -c1 8.8.8.8")
+                self.os_conn.disassociate_floating_ip(
+                    server, self.floating_ip, use_neutron=True)
+                res2 = vm_remote.execute("date")
+
+        assert (0 == res1['exit_code'],
```
Comment (gdyuldin) [inline mos_tests/neutron/python_tests/test_floating_ip.py]: This assert will never fail (not empty tuple is True)

File: mos_tests/environment/ssh.py (hunk 1 above)
Comment (alexgrom9393) [inline mos_tests/environment/ssh.py]: Done

File: mos_tests/neutron/python_tests/base.py (hunk 2 above)
Comment (alexgrom9393) [inline mos_tests/neutron/python_tests/base.py]: Done

File: mos_tests/neutron/python_tests/test_floating_ip.py (hunk 3 above)
Comment (alexgrom9393) [inline mos_tests/neutron/python_tests/test_floating_ip.py]: We decided that there is no sense in this additional check of message.
So it was deleted.
Done

File: mos_tests/neutron/python_tests/test_floating_ip.py (hunk 4 above)
Comment (alexgrom9393) [inline mos_tests/neutron/python_tests/test_floating_ip.py]: Moved to common cleanup function for floating_ips in os_actions.
Done

File: mos_tests/neutron/python_tests/test_floating_ip.py (hunk 5 above)
Comment (alexgrom9393) [inline mos_tests/neutron/python_tests/test_floating_ip.py]: Done

File: mos_tests/neutron/python_tests/test_floating_ip.py [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+        Steps:
+            1. Create network net01, subnet net01__subnet with CIDR 10.1.1.0/24
+            2. Create new security group sec_group1
+            3. Add Ingress rule for TCP protocol to sec_group1
+            4. Boot vm1 net01 with sec_group1
+            5. Associate floating IP for vm1
+            6. Go to vm1 with ssh and floating IP
+            7. Without stopping ssh-connection dissociate floating ip from vm
+            8. Check that connection is stopped
+            9.Try to go to vm1 with ssh and floating IP
```
Comment (TimurNurlygayanov) [inline mos_tests/neutron/python_tests/test_floating_ip.py]: 9.Try  > 9. Try

Comment (alexgrom9393) [inline mos_tests/neutron/python_tests/test_floating_ip.py]: Done.

**Pull request merged**